modules/top-level/plugins/combine-plugins.nix

```diff
@@ -0,0 +1,103 @@
+{ lib, pkgs }:
+let
+  inherit (import ./utils.nix lib)
+    normalizePlugin
+    normalizePluginList
+    removeDependencies
+    ;
+
+  # Combine a list of plugins into a single normalized plugin "pack"
+  __mkPluginPack =
+    { pluginsToCombine, pathsToLink }:
+    let
+      # Every plugin has its own generated help tags (doc/tags)
+      # Remove them to avoid collisions, new help tags
+      # will be generate for the entire pack later on
+      overriddenPlugins = map (
+        plugin:
+        plugin.plugin.overrideAttrs (prev: {
+          nativeBuildInputs = lib.remove pkgs.vimUtils.vimGenDocHook prev.nativeBuildInputs or [ ];
+          configurePhase = ''
+            ${prev.configurePhase or ""}
+            rm -vf doc/tags'';
+        })
+      ) pluginsToCombine;
+
+      # Gather python 3 dependencies from every plugins
+      python3Dependencies =
+        let
+          deps = map (p: p.plugin.python3Dependencies or (_: [ ])) pluginsToCombine;
+        in
+        ps: builtins.concatMap (f: f ps) deps;
+
+      # Combined plugin
+      combinedPlugin = pkgs.vimUtils.toVimPlugin (
+        pkgs.buildEnv {
+          name = "plugin-pack";
+          paths = overriddenPlugins;
+          inherit pathsToLink;
+          # Remove empty directories and activate vimGenDocHook
+          postBuild = ''
+            find $out -type d -empty -delete
+            runHook preFixup
```

Comment:
I know this is out of scope for this PR, but I find it very non-obvious that running `preFixup` within `postBuild` is "activate vimGenDocHook". This feels very very hacky.

Reply:
Do you know the correct approach? If not would you consider adding some additional notes/comments/fixmes?